### PR TITLE
Fixed raceAttempt: losing fiber will now get interrupted on success too

### PR DESCRIPTION
**Pull request by lbialy**
As in description - behaviour of `raceAttempt` is inconsistent with all other `race*` variants - it doesn't interrupt losing fiber if winning fiber finishes successfully. This PR fixes this.

### CODE REVIEW

**Comment by lbialy** [comment]
Maybe a test case should be added?

**Comment by NeQuissimus** [comment]
I would love a test for this :-)

**Comment by lbialy** [comment]
There's a test for `raceAttempt` that tests this:
```
  def testRaceAttemptInterruptsLoser =
    unsafeRun(for {
      s      <- Promise.make[Nothing, Unit]
      effect <- Promise.make[Nothing, Int]
      winner = s.await *> IO.fromEither(Left(new Exception))
      loser  = IO.bracket(s.succeed(()))(_ => effect.succeed(42))(_ => IO.never)
      race   = winner raceAttempt loser
      _      <- race.either
      b      <- effect.await
    } yield b) must_=== 42
```
Surely you can see why the oversight of lack of interruption of loser could happen. I will split this into two tests, one for error, one for result.

**Comment by lbialy** [comment]
```
[error]     x raceAttempt interrupts loser on success
[error]  TIMEOUT: 60 seconds (TestRuntime.scala:18)
```
So far so good, fails when I revert my changes...

**Comment by lbialy** [comment]
How in seven hells did this pass CI? My change breaks two tests in RTSSpec?

**Comment by lbialy** [comment]
Ok, it seems RTSSpec is flaky, it just passed successfully after sbt reload :thinking:

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 1]
```diff
@@ -250,8 +250,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
    */
   final def raceAttempt[R1 <: R, E1 >: E, A1 >: A](that: ZIO[R1, E1, A1]): ZIO[R1, E1, A1] =
     raceWith(that)(
-      { case (l, f) => l.fold(f.interrupt *> ZIO.halt(_), ZIO.succeed) },
-      { case (r, f) => r.fold(f.interrupt *> ZIO.halt(_), ZIO.succeed) }
+      { case (l, f) => l.fold(f.interrupt *> ZIO.halt(_), f.interrupt *> ZIO.succeed(_)) },
```
Comment (regiskuckaertz) [inline core/shared/src/main/scala/zio/ZIO.scala]: Small change for readability:
```suggestion
      { case (l, f) => f.interrupt *> l.fold(ZIO.halt, ZIO.succeed) },
```

File: core/shared/src/main/scala/zio/ZIO.scala [hunk 2]
```diff
@@ -250,8 +250,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
...
+      { case (r, f) => r.fold(f.interrupt *> ZIO.halt(_), f.interrupt *> ZIO.succeed(_)) }
```
Comment (regiskuckaertz) [inline core/shared/src/main/scala/zio/ZIO.scala]: ditto
```suggestion
      { case (r, f) => f.interrupt *> r.fold(ZIO.halt, ZIO.succeed) }
```

**Comment by jdegoes** [comment]
@lbialy Looks great. Thanks for your contribution and congratulations on your first ZIO pull request!

@regiskuckaertz @kaishh Thanks for the reviews...

giphy

**Comment by lbialy** [comment]
It's actually the second one :laughing:

**Comment by jdegoes** [comment]
@lbialy Oops, you get extra fireworks I suppose. 😉

**Pull request merged**